src/psyclone/psyir/nodes/assignment.py

```diff
@@ -73,6 +81,15 @@ def _validate_child(position, child):
         '''
         return position < 2 and isinstance(child, DataNode)
 
+    def __eq__(self, other):
+        '''
+        :param Any other: the object to check equality to.
+
+        :returns: whether this type is equal to the 'other' type.
+        :rtype: bool
+        '''
+        return super().__eq__(other) and self.is_pointer == other.is_pointer
```

Comment:
Better to do the `is_pointer` comparison first as that will be cheaper (presumably)?

Reply:
It is in purpose, as first we need to compare the type to guarantee that they both have the "is_pointer" attribute (as we can compare to anything else). All node equals should be implemented in this style. I added a test that would fail if the order was reversed.

Reply:
Great.